circuits/cpp/src/aztec3/circuits/rollup/test_utils/utils.cpp

```diff
@@ -130,6 +138,62 @@ BaseRollupInputs base_rollup_inputs_from_kernels(std::array<KernelData, 2> kerne
     baseRollupInputs.new_commitments_subtree_sibling_path =
         get_sibling_path<PRIVATE_DATA_SUBTREE_INCLUSION_CHECK_DEPTH>(private_data_tree, 0, PRIVATE_DATA_SUBTREE_DEPTH);
 
+    // Update public data tree to generate sibling paths: we first set the initial public data tree to the result of all
+    // state reads and old_values from state transitions. Note that, if the right tx reads or writes an index that was
+    // already processed by the left one, we don't want to reflect that as part of the initial state, so we skip those.
+    std::set<size_t> visited_indices;
+    for (size_t i = 0; i < 2; i++) {
+        for (auto state_read : kernel_data[i].public_inputs.end.state_reads) {
+            auto leaf_index = size_t(uint256_t(state_read.leaf_index));
```

Comment:
Why do you convert to `size_t` here? Aren't you truncating most of the indices to 64 bits?

Reply:
Good catch! It was a leftover from using a function from the append-only MemoryTree, which uses size_t. 

Unfortunately, changing this throws an error in using the unordered_set, since it requires a default constructor for the uint256_t type which is apparently not available. So I'm reverting the unordered_set change, so we can use correct indices.